crates/rustc_codegen_spirv/src/builder/spirv_asm.rs

```diff
@@ -304,8 +304,26 @@ impl<'cx, 'tcx> Builder<'cx, 'tcx> {
             }
             .def(self.span(), self),
             Op::TypeArray => {
-                self.err("OpTypeArray in asm! is not supported yet");
-                return;
+                let count = inst.operands[1].unwrap_id_ref();
+                let get_count_ty = || -> Option<Word> {
+                    let emit = self.emit();
+                    let func = &emit.module_ref().functions[emit.selected_function()?];
+                    let insts = &func.blocks[emit.selected_block()?].instructions;
+                    let inst = insts.iter().find(|i| i.result_id == Some(count))?;
+                    inst.result_type
+                };
```

Comment:
Where is this tested/exercised? I'm worried that the count constant wouldn't end up in the current function, but as a global. Or at least I didn't think constants are valid inside functions.

Reply:
It was required to be implemented for - oh, hold on, I actually deleted that bit of the `asm!`, replaced it with a `typeof*{result}`, I'll just revert this because it was partially hacky (but yeah, the constant does show up in the current function, even though it's not valid there - when rspirv reparses the module for linking, it reshuffles everything to its proper position. whee, hacks.)

Reply:
> when rspirv reparses the module for linking, it reshuffles everything to its proper position

Oh dear that explains things to me that subtly confused me months ago (but not enough for me to ask anyone or try to track it down in the code.